src/os.cc

```diff
@@ -239,7 +230,11 @@ OS::HeapMemoryRange OS::get_heap_memory_range() {
     // be the last MAX_HEAP of the address space.
     single_range_.address = reinterpret_cast<void*>(-static_cast<word>(MAX_HEAP + TOIT_PAGE_SIZE));
   } else {
+#if defined(TOIT_LINUX) and defined(BUILD_64)
```

Comment:
I'm not sure I fully understand the logic here. Maybe add comments on why we do one thing on 64-bit Linux and another elsewhere?